specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json

```diff
@@ -313,7 +313,7 @@
     },
     "LUISParameters": {
       "description": "This is a set of request parameters for LUIS Generally Available projects.",
```

Comment:
Are you intentionally keeping it all caps in the docs?

Reply:
Yes. LUIS in the doc is intended because that is the name of our GA product.

Reply:
I am not sure if this will break code generation. I think No because CLU Authoring made the same change. @heaths ?

Reply:
It will - most languages are case-sensitive - but you're breaking a preview. Being consistent across COGS Language pillar before we GA is worth it, IMO.

Reply:
I agree it's worth it. This is our final decision to be consistent across at least CLU authoring and runtime, as Abbas is correcting these in Authoring Swagger. Do I need to update anything in this PR, like renaming `LUISParameters` to `LuisParameters`?

Reply:
FWIW, this seems to be based on the guidelines saying acronyms should be PascalCase, but I believe that was only intended for models and property names. I had a comment in another PR and tagged a few people for follow-up. If you have the time and would prefer "LUIS", we might want to wait for an answer.

Reply:
Sounds good. I will wait for it. The back-end service needs more time to change, but this renaming to the class names does not require back-end changes. So we should be good.

Reply:
Hi @heaths , is there any update on this? I can update LUIS to Luis. I am totally OK with that.

Reply:
For model names and property names, this should be `Luis`. I was referring to property values, which you also changed. @markweitzel @mikekistler do I have that correct - only model/property names should use "Luis" while enum values, for example, can use "LUIS", or should they all use "Luis". Specs aren't crystal clear.

Reply:
We have not included a convention for enum values in the guidelines _yet_, but the issue has come up previously and after discussion the consensus was that enum values should also be camel case.  I captured this in this issue:

https://github.com/microsoft/api-guidelines/issues/342

Reply:
I'll comment on that, but I would assume that acronyms would be all lowercase then e.g., "luis".

Reply:
So are we going to use all lowercase for all enum values?

Reply:
The proposal -- not yet codified in the API Guidelines -- is that enum values should be camel case (specifically _lower_ camel case -- i.e. the first letter is always lower case). And acronyms should be treated as "words" in this context, so in this particular case the recommendation would be "luisParameters".

Reply:
@heaths If we change the model name as `luisParameters`, will there be any code generation issue in some languages?

Reply:
No. The languages' code generators will use the proper naming conventions e.g., Python will generate `luis_parameters` and C# will generate `LuisParameters`.

Reply:
Sounds good. I will update the model name to `luisParameters`.

For enum values, we will use `luis`, `orchestration`, `conversation`, and `questionAnswering`, `nonLinked`. Does this look good to you?

Reply:
Yes. I would also update the title of this PR since it will be saved to history forever. Might as well make the title right for easier troubleshooting.

Reply:
Hi @moabba, after some discussion, the board recommended to use camel case for enum values. I am sorry for the back and forth. Can you update the authoring to use the same values?

For enum values, we will use `luis`, `orchestration`, `conversation`, `questionAnswering`, `nonLinked`.

Reply:
The authoring uses camelCase for all the enum values. We had to change the project type casing to PascalCase to match both CLU and CT prediction.

Text Analytics use a lot of different variants, but the task kinds (project types in authoring) are PascalCase. I would recommend keeping this convention for consistency.

If we had enough extra time, I would've asked TA to use camelCase, and then use camelCase for the enum values you're mentioning. But unfortunately, we don't have this time (GA is in 2 weeks).

Reply:
@heaths , I totally agree with Abbas. Both TA and CLU runtime are using PascalCase right now, it would require all services to update in a short time before GA. Are we OK with PascalCase right now?

Reply:
That's fine, but I would make it consistent between your inference and authoring swaggers.

Remember, the guidelines as currently written were only about model and property names. If there's no time to change those I suppose that's fine - our SDKs should still do the right thing (we can fix them if not); however, the property values weren't necessary to change as currently documented. Mike only mentioned that we were going to update the guidelines to recommend camelCase.

Reply:
After this PR get merged, the inference runtime and authoring will be consistent. I guess that means there is no more update to this PR. Heath, if you can approve it then we can merge it.

Reply:
I'm confused. I thought the models you renamed to `LuisParameters` would need to be reverted to `LUISParameters` then. Or are they already? This has gone back and forth enough I have to admit I'm confused what the desired casing is for:

* Models
* Model properties
* Property values e.g., enum values

Could you clarify?

Reply:
Sorry for the confusion. `LUISParameters` is unique to the runtime. This PR means to align enum values with the authoring side. 

The change on model name `LUISParameters` is to simply unify what we used in enum values. Do you suggest I revert the model name back and only change enum values in this PR?

Reply:
The recommendations are (or will be):

* Models are PascalCase
* Model properties are camelCase
* Property / enum values are camelCase

I would try to align to those as much as you can / are comfortable with now. But please let me know in a comment. I would like to review all the swaggers for GA to see if we're consistent, or is consistency not achievable in the necessary timeframe?